### ISSUE TITLE
Support generating nullable array elements

### ISSUE THREAD

**Issue by QZone99** [apple/swift-openapi-generator#444]
Error: 
```
Client error - cause description: 'Unknown', underlying error: DecodingError: valueNotFound Double
```
```
Cannot get unkeyed decoding container -- found null value instead (underlying error: <nil>)
```

openapi.yaml
```
openapi: 3.0.3
info:
  title: MyService
  version: 1.0.0
servers:
  - url: 'https://api.myservice.com'
    description: myservice
paths:
  '/data/{id}':
    get:
      operationId: data
      parameters:
        - name: id
          required: true
          in: path
          description: The id.
          schema:
            type: string
      responses:
        '200':
          description: A success response with a MyData.
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/MyData'
components:
  schemas:
    MyData:
      type: object
      required: []
      properties:
        points:
          type: object
          required: []
          properties:
            result:
              type: array
              items:
                type: object
                required: []
                properties:
                  one:
                    type: array
                    items:
                      type: number
                      nullable: true
```
```
% sw_vers                                                       
ProductName:        macOS
ProductVersion:     14.1.2
BuildVersion:       23B92
```
My schema requires that the type of array's items (numbers) need to be nullable (or `nil`).

**Comment by czechboy0** [comment]
It seems the payload contains a Double value instead of an Array. Please double check your schema, this doesn't appear like a bug in Swift OpenAPI Generator. If you identify unexpected generator behavior, please add more details.

**Comment by QZone99** [comment]
Versions: `1.0.0-alpha.1`

The example of JSON payload excerpt that caused the issue:
```
{
    "one": [1.1, 1.2, 1.3, null, 1.5, 1.6, null, 1.8]
}
```

The item is defined as type`number` and `nullable: true`:
```
one:
  type: array
  items:
    type: number
    nullable: true
```
The generated code is:
```
public init(one: [Swift.Double]? = nil)
```
But I need:
```
public init(one: [Swift.Double?]? = nil)
```
The error indicates that 
```
Cannot get unkeyed decoding container -- found null value instead (underlying error: <nil>)
```

**Comment by czechboy0** [comment]
Fixed in [1.1.0](https://github.com/apple/swift-openapi-generator/releases/tag/1.1.0).
